src/App.tsx

```diff
@@ -1,9 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
+import { Movies } from './types/movies';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const handleChangeFilterText = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+  };
+
+  const visibleMovies = (moviesArray: Movies[], filter: string): Movies[] => {
+    const moviesCopy = [...moviesArray];
```

Comment:
[checklist](https://github.com/mate-academy/react_movies-list-filter/blob/master/checklist.md): you don't need to create copy because filter is non-mutating method so it will return altered copy of moviesArray